scheduler/src/cook/mesos/util.clj

```diff
@@ -150,15 +150,21 @@
 (def ^:const default-job-priority 50)
 
 (defn same-user-task-comparator
-  "Comparator to order same user's tasks"
-  [task1 task2]
-  (letfn [(task->feature-vector [task]
+  "Comparator to order same user's tasks
+
+   If consider-backfilled-jobs? is true, then we treat jobs which were backfilled as being the lowest priority"
+  [consider-backfilled-jobs?]
```

Comment:
@wyegelwel in the vector to be compared, the first element is 1 iff backfilled? matters and the job is backfilled, 0 otherwise.   0 sorts before 1 so non-backfilled jobs will be sorted to the beginning of the list...  

To respond vaguely and in general to a lot of what you wrote, yes, this is complicated :)